### DISCUSSION TITLE
Help loading library with System.js when using Webview

### DISCUSSION THREAD

**Discussion by jardg** [react-hook-form/react-hook-form#4755] [Q&A] [answered]
Hi guys. I'm loading the library using system.js with the form.umd file. This is the file I use: https://www.unpkg.com/react-hook-form@5.7.2/dist/react-hook-form.umd.js . Then the library is configured as an external dependency on webpack config(I'm using single-spa.js to load microfrontends so I share these libraries to reduce network latency and bundles)
On  website works great but then I have a react-native application that loads that website using React Native WebView and I receive the following error:  SyntaxError: Unexpected token (
That happens on the following line  var validateField = async (fieldsRef, validateAllFieldCriteria, { ref, ref: { type, value, name }, options, required, maxLength, minLength, min, max . I'm guessing the WebView doesn't process the async well.
I tried loading babel polyfill before but doesn't have any effect.  Also tried to use the library with IE compatibility with https://www.unpkg.com/react-hook-form@5.7.2/dist/react-hook-form.ie11.js But then says Exports are not defined since I'm using System.js
Could you please guide me a bit on how to resolve this problem or what I'm doing wrong?Thank you very much

**Comment by bluebill1049** [comment, marked as answer]
which version are you running? looks like web code go into the native for some reason.

**Reply by jardg** [reply]
Hi!. In the React Native app I'm using:

"react": "^16.13.1",
    "react-native": "0.63.2",
    "react-native-webview": "^11.3.2"

While the server web app uses:

    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    single-spa:5.8.2
    SystemJS: 6.8.0
   react-hook-form: 5.7.2

In the server I load all the libraries with SystemJS. They are marked as externals in webpack config. For react-hook-form I use the file react-hook-form.umd.js in https://www.unpkg.com/browse/react-hook-form@5.7.2/dist/ . So I presume the native app is loading all the libraries with system.js too but fails to understand the async syntax well

**Reply by bluebill1049** [reply]
oh v5, there may be not incorrect env check in there. Any chance you can upgrade to v6? they r quite similar. V5 is no longer getting maintained as well.

**Reply by jardg** [reply]
Sure. I'm gonna try and see if I can make it work

**Reply by jardg** [reply]
I tried 6.13.0 . Same issue with async in line 419 var validateField = async (fieldsRef, valida ...
I used this script https://www.unpkg.com/react-hook-form@6.13.0/dist/index.umd.development.js

Maybe I need babel/preset-env on react native app?

**Reply by bluebill1049** [reply]
so the syntax is to do with async, hmmm... yea, i am not too sure on this, but try with babel.

**Comment by jardg** [comment]
Ended up applying @babel/plugin-transform-async-to-generator in the online page https://babeljs.io/ to the react-hook-form umd file and solved my issue. I'm having more problems but not are not related to this library. Not ideal but I don't know any other way to apply babel to a file that I load with system.js.
Thanks a lot for the help since it made me think the solution in another way.

**Reply by bluebill1049** [reply]
No worries buddy.
